File: src/services/EmergencyRecordingService.ts
// PLACEHOLDER: This service defines the full API for emergency video recording.
// All methods are no-ops until the recording feature is enabled.
// When enabled, start() will launch a native foreground service that records
// camera + microphone and uploads chunks to Firebase Storage, and family members
// will receive a viewer link via the existing notification channel.

const EmergencyRecordingService = {
  // PLACEHOLDER
  async start(_userId: string, _groupId: string): Promise<void> {
    console.log('[EmergencyRecordingService] PLACEHOLDER: recording not yet implemented');
  },

  // PLACEHOLDER
  async stop(): Promise<void> {
    console.log('[EmergencyRecordingService] PLACEHOLDER: stop called');
  },

  // PLACEHOLDER
  isRecording(): boolean {
    return false;
  },
};

export default EmergencyRecordingService;

File: android/app/src/main/java/com/familysafety/EmergencyRecordingService.java
package com.familysafety;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

// PLACEHOLDER: This service is declared but not functional.
// When the recording feature is enabled, this will become a foreground service
// that captures camera/microphone and uploads chunks to Firebase Storage.
public class EmergencyRecordingService extends Service {

    private static final String TAG = "EmergencyRecording";

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d(TAG, "PLACEHOLDER: Emergency recording service started — no recording in progress");
        return START_STICKY;
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}
